### DISCUSSION TITLE
Hierarchical tags

### DISCUSSION THREAD

**Discussion by meredrica** [zk-org/zk#479] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

It would be great to have tags that provde a hierarchy, like `#todo/read` and `#todo/write` that get listed when you search with `zk list --tags todo`

**Comment by marty-oehme** [comment]
If I am not missing something I think this is already possible, at least for how I use the program.
I just tried it in a little experimental notebook dir and it worked quite well.
Be aware that I am using colon-delimited tags (:tag:) but I don't think this is crucial for the function.

Let's take your example, I can have some notes tagged with `:todo/read:`, some with `:todo/write:` and some just `:todo:`.

To list all the notes that are still to be read, I use:

```bash
zk list --tag "todo/read"
```

This shows _only_ those exactly tagged notes (not `/write` and not `todo` only).
Similarly of course for matching on `todo/write`.

```bash
zk list --tag "todo"
```

This matches only notes that have been tagged _exactly_ `todo`, not those to be read or written.

```bash
zk list --tag "todo*"
```

This gives me all the notes under the todo 'hierarchy', whether to be read, written or just general todos.

```bash
zk list --tag "todo/*"
```

And this finally gives me all the notes to be either read or written still, but _not_ those just tagged with a general todo.

The one thing that could be improved is showing which of the specific tags a note belongs to if we are showing the glob results of multiple but I think this already goes a long way to providing quasi-hierarchical tags.

**Reply by meredrica** [reply]
I honestly didn't realize that I can just use a wildcard 🤦. Thanks for pointing this out, I'll close the issue.
